packages/core/test-helpers/core-test-helpers-so-type-serializer/src/extract_migration_info.test.ts

```diff
@@ -158,4 +158,108 @@ describe('extractMigrationInfo', () => {
       });
     });
   });
+
+  describe('modelVersions', () => {
```

Comment:
nit: should we add a test for a mixed migration (legacy and new one)?

Reply:
> a mixed migration

Is that even supported? How would that work? I can't imagine supporting BWC in that scenario would be easy.

Reply:
> Is that even supported?

Yeah it is supported (see https://github.com/elastic/kibana/issues/150301 - we don't really have a choice for on-prem). It's just that you won't be able to down-migrate lower than model versions.